tools/common/schema/updatetask.go

```diff
@@ -152,9 +152,22 @@ func (task *UpdateTask) execStmts(ver string, stmts []string) error {
 	task.logger.Debug(fmt.Sprintf("---- Executing updates for version %v ----", ver))
 	for _, stmt := range stmts {
 		task.logger.Debug(rmspaceRegex.ReplaceAllString(stmt, " "))
-		e := task.db.Exec(stmt)
-		if e != nil {
-			return fmt.Errorf("error executing statement:%v", e)
+		err := task.db.Exec(stmt)
+		if err != nil {
+			// To make schema update idempotent, we need to handle error when retry on previous partially succeeded update attempt.
+			// There are 3 cases that will be handled:
+			// 1) Add table that already exists: Cassandra returns error of type *gocql.RequestErrAlreadyExists.
+			// 2) Add column that already exists: Cassandra returns error of type *gocql.errorFrame with message contains "already exists".
+			// 3) Drop column that is not found: Cassandra returns error of type *gocql.errorFrame with message contains "not found".
+			// gocql.errorFrame is a private error type that we cannot reference.
+			var alreadyExistsErr *gocql.RequestErrAlreadyExists
+			if errors.As(err, &alreadyExistsErr) ||
+				strings.Contains(err.Error(), "already exists") ||
+				strings.Contains(err.Error(), "not found") {
+				task.logger.Warn("Duplicate update, most likely due to previous partially succeeded update attempt. Ignoring it and continue.", tag.Error(err))
+			} else {
+				return fmt.Errorf("error executing statement: %w", err)
+			}
```

Comment:
Maybe try this to resolve the lint

```suggestion
			if errors.As(err, &alreadyExistsErr) ||
				strings.Contains(err.Error(), "already exists") ||
				strings.Contains(err.Error(), "not found") {
				task.logger.Warn("Duplicate update, most likely due to previous partially succeeded update attempt. Ignoring it and continue.", tag.Error(err))
				continue
			}
			return fmt.Errorf("error executing statement: %w", err)
```